node/api/api.go

```diff
@@ -162,3 +184,77 @@ func WriteJSON(w http.ResponseWriter, obj interface{}) {
 func WriteSuccess(w http.ResponseWriter) {
 	w.WriteHeader(http.StatusNoContent)
 }
+
+func GetPaginationDefaultRequest(req *http.Request) PaginationRequest {
+	return GetPaginationRequest(req, "start", "limit")
+}
+
+func GetPaginationRequest(req *http.Request, startQueryParam string, limitQueryParam string) PaginationRequest {
```

Comment:
I would remove the `PaginationRequest` struct and simply return the  `startIndex`, `limit` and an error that indicates a parsing problem. That way the method gets significantly shorter since we don't always construct the `PaginationRequest` object and we can return `build.ExtendErr`-extended errors that indicate which parameter couldn't be parsed.

Reply:
If you still want to be able to differentiate between a "no pagination params specified" error and a "parsing error" you can declare a custom error for which you can check in the calling method.

Reply:
May I also suggest a different name for the method? `ParsePaginationParams` might be better after getting rid of the `PaginationRequest` struct.